File: front/src/client/BaseClient.js
class BaseClient {
    static addPagination(url, limit, page) {
    const params = [];

    if (typeof limit === 'number' || typeof limit === 'string') {
        params.push(`limit=${encodeURIComponent(limit)}`);
    }
    if (typeof page === 'number' || typeof page === 'string') {
        params.push(`page=${encodeURIComponent(page)}`);
    }

    if (params.length > 0) {
        const separator = url.includes('?') ? '&' : '?';
        const resultString = params.join('&');

        return url + separator + resultString;
    } else {
        return url;
    }
  }
}

export default BaseClient

File: src/client/BaseClient.py
class BaseClient:
    @staticmethod
    def _add_pagination(url: str, limit: int, page: int):
        params = []

        if limit:
            params.append(f"limit={limit}")
        if page:
            params.append(f"page={page}")

        if params:
            separator = "&" if "?" in url else "?"
            result_string = separator + "&".join(params)
            return url + result_string
        else:
            return url
